src/scripts/domParsers/youtubeVideoExtractor.ts

```diff
@@ -0,0 +1,75 @@
+import {DomUtils} from "./domUtils";
+import {VideoExtractor} from "./videoExtractor";
+
+import {Utils} from "../utils";
+
+export class YoutubeVideoExtractor implements VideoExtractor {
+	private youTubeWatchVideoBaseUrl = "https://www.youtube.com/watch";
+	private youTubeVideoIdQueryKey = "v";
+	private dataOriginalSrcAttribute = "data-original-src";
+
+	/**
+	 * Return the ID of the video in the YouTube URL as an array
+	 */
+	getVideoIds(youTubeUrl: string, pageContent: string): string[] {
+		if (Utils.isNullOrUndefined(youTubeUrl)) {
+			return;
+		}
+
+		let youTubeId;
+		if (Utils.getPathname(youTubeUrl).indexOf("/watch") === 0) {
+			youTubeId = Utils.getQueryValue(youTubeUrl, this.youTubeVideoIdQueryKey);
+			if (Utils.isNullOrUndefined(youTubeId)) {
+				return;
+			}
+		}
+
+		if (Utils.getPathname(youTubeUrl).indexOf("/embed") === 0) {
+			let youTubeIdMatch = youTubeUrl.match(/youtube\.com\/embed\/(\S+)/);
```

Comment:
Some of the videos on KhanAcademy have hyphens in their data-ids, and `\w` does not cover those.